spec/immutable-data-structures.html

```diff
@@ -6,27 +6,590 @@ <h1>Immutable Data Structures</h1>
+
+        <p>This property has the attributes { [[Writable]]: false, [[Enumerable]]: false, [[Configurable]]: false }.</p>
       </emu-clause>
     </emu-clause>
   </emu-clause>
 
   <emu-clause id="sec-tuple-objects">
     <h1>Tuple Objects</h1>
-    <p>TBD</p>
     <emu-clause id="sec-tuple-constructor">
       <h1>The Tuple Constructor</h1>
-      <p>TBD</p>
-      <emu-clause id="sec-tuple-constructor-tuple-value">
-        <h1>Tuple ( _value_ )</h1>
-        <p>TBD</p>
+      <p>The Tuple constructor:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%Tuple%</dfn>.</li>
+        <li>is the initial value of the *"Tuple"* property of the global object.</li>
+        <li>creates and initializes a new Tuple object when called as a function.</li>
+        <li>is not intended to be used with the `new` operator or to be subclassed. It may be used as the value of an *extends* clause of a class definition but a *super* call to the Tuple constructor will cause an exception.</li>
+      </ul>
+      <emu-clause id="sec-tuple-items">
+        <h1>Tuple ( ..._items_ )</h1>
+        <p>When the `Tuple` function is called, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is not *undefined*, throw a *TypeError* exception.
+          1. Let _elements_ be a List containing the arguments passed to this function.
+          1. Let _length_ to be the number of elements in _elements_.
+          1. For each element _e_ of _elements_,
+            1. If Type(_e_) is Object, throw a *TypeError* exception.
+          1. Let _tuple_ be a new Tuple value whose [[Sequence]] is _elements_.
+          1. Return _tuple_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+    <emu-clause id="sec-properties-of-the-tuple-constructor">
+      <h1>Properties of the Tuple Constructor</h1>
+      <p>The Tuple constructor:</p>
+      <ul>
+        <li>has a [[Prototype]] internal slot whose value is %Tuple.prototype%.</li>
+        <li>has the following properties:</li>
+      </ul>
+      <emu-clause id="sec-tuple.istuple">
+        <h1>Tuple.isTuple ( _arg_ )</h1>
+        <p>The *isTuple* function takes one argument _arg_, and performs the following steps:</p>
+        <emu-alg>
+          1. Return ? IsTuple(_arg_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.from">
+        <h1>Tuple.from ( _items_ [ , _mapFn_ [, _thisArg_ ] ] )</h1>
+        <p>When the `from` method is called with argument _items_ and optional arguments _mapfn_ and _thisArg_, the following steps are taken:</p>
+        <emu-alg>
+          1. If _mapfn_ is *undefined*, let _mapping_ be *false*.
+          1. Let _list_ be a new empty List.
+          1. Else,
+            1. If IsCallable(_mapfn_) is *false*, throw a *TypeError* exception.
+            1. Let _mapping_ be *true*.
+          1. Let _usingIterator_ be ? GetMethod(_items_, @@iterator).

```

Comment:
could this whole function use AddEntriesFromIterable instead of duplicating Array.from?

Reply:
At least the first half can, since the second half handles array-likes.